workflow/util/util.go

```diff
@@ -814,7 +814,13 @@ func retryWorkflow(ctx context.Context, kubeClient kubernetes.Interface, hydrato
 			return nil, errors.InternalErrorf("Workflow cannot be retried with node %s in %s phase", node.Name, node.Phase)
 		}
 		if node.Type == wfv1.NodeTypePod {
-			podName := PodName(wf.Name, node.Name, node.TemplateName, node.ID)
+			var templateName string
```

Comment:
I see similar code here, but the template name format is different in the case of TemplateRef. Are we sure this is correct?

https://github.com/argoproj/argo-workflows/blob/7739256d4b28995e5af33d2e54c85322ed5dfcba/cmd/argo/commands/get.go#L532-L537

Reply:
v2 pod name doesn't contain WorkflowTemplate's name (=`TemplateRef.name`).

![image](https://user-images.githubusercontent.com/23391543/144263036-a57cf765-5882-4d57-a859-a6d7ade843e6.png)

It consists of [workflow name]-[template name]-[hash] .

---

Workflow
![image](https://user-images.githubusercontent.com/23391543/144263149-912608eb-af4c-485b-a987-13fbd3fe8e11.png)

WorkflowTemplates
![image](https://user-images.githubusercontent.com/23391543/144263226-bc0d94a2-cd21-4200-8ee8-605c0effdd8a.png)

each full manifest is in #7315

Reply:
Hey @mikutas - I'd recommend using the utility function for getting the template name: https://github.com/argoproj/argo-workflows/blob/master/ui/src/app/shared/pod-name.ts#L49

That utility function could use a test though. Feel free to add one in this PR